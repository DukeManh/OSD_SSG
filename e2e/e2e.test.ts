import run, { build } from './run';

function sortConsoleLogs(out: string) {
  const messages = out.split(/\r?\n/).sort();
  return messages.join('\n');
}

describe('end to end test', () => {
  beforeAll(async () => {
    try {
      await build();
    } catch {
      process.exit(1);
    }
  }, 15000);
  test('no input file path given', async () => {
    const { stderr, stdout, exitCode } = await run('');
    expect(sortConsoleLogs(stderr)).toMatchSnapshot();
    expect(exitCode).toEqual(1);
    expect(stdout).toEqual('');
  });

  test('only set input file path', async () => {
    const { stderr, stdout, exitCode } = await run('-i assets/text');
    expect(stderr).toEqual('');
    expect(exitCode).toEqual(0);
    expect(sortConsoleLogs(stdout)).toMatchSnapshot();
  });

  test('use relative option', async () => {
    const { stderr, stdout, exitCode } = await run('-i assets/text -e');
    expect(stderr).toEqual('');
    expect(exitCode).toEqual(0);
    expect(sortConsoleLogs(stdout)).toMatchSnapshot();
  });

  test('use recursive option', async () => {
    const { stderr, stdout, exitCode } = await run('-i assets/text -r');
    expect(stderr).toEqual('');
    expect(exitCode).toEqual(0);
    expect(sortConsoleLogs(stdout)).toMatchSnapshot();
  });

  test('use both recursive and relative option', async () => {
    const { stderr, stdout, exitCode } = await run('-i assets/text -e -r');
    expect(stderr).toEqual('');
    expect(exitCode).toEqual(0);
    expect(sortConsoleLogs(stdout)).toMatchSnapshot();
  });

  test('use config file', async () => {
    const { stderr, stdout, exitCode } = await run('-c assets/config/config.json');
    expect(stderr).toEqual('');
    expect(exitCode).toEqual(0);
    expect(sortConsoleLogs(stdout)).toMatchSnapshot();
  });

  test('use invalid css file', async () => {
    const { stderr, stdout, exitCode } = await run('-i assets/text -s src/index.ts');
    expect(stderr).toMatchSnapshot();
    expect(exitCode).toEqual(1);
    expect(stdout).toEqual('');
  });

  test('use invalid language ', async () => {
    const { stderr, stdout, exitCode } = await run('-i assets/text -l alien-lang');
    expect(stderr).toMatchSnapshot();
    expect(exitCode).toEqual(1);
    expect(stdout).toEqual('');
  });

  test('use input option on a file', async () => {
    const { stderr, stdout, exitCode } = await run('-i assets/text/The\\ Naval\\ Treaty.txt');
    expect(stderr).toEqual('');
    expect(exitCode).toEqual(0);
    expect(sortConsoleLogs(stdout)).toMatchSnapshot();
  });

  test('use recursive option on a file', async () => {
    const { stderr, stdout, exitCode } = await run('-i assets/text/The\\ Naval\\ Treaty.txt -r');
    expect(stderr).toEqual('');
    expect(exitCode).toEqual(0);
    expect(sortConsoleLogs(stdout)).toMatchSnapshot();
  });
});
