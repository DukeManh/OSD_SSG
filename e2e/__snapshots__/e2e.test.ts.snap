// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`end to end test no input file path given 1`] = `"Missing required argument: input"`;

exports[`end to end test only set input file path 1`] = `
"Created 'build/The Naval Treaty.html'
Created 'build/The Red Headed League.html'"
`;

exports[`end to end test use both recursive and relative option 1`] = `
"Created 'build/Markdown/Open Source.html'
Created 'build/The Naval Treaty.html'
Created 'build/The Red Headed League.html'
Created 'build/a/Silver Blaze.html'
Created 'build/a/b/The Adventure of the Six Napoleans.html'
Created 'build/c/The Adventure of the Speckled Band.html'"
`;

exports[`end to end test use config file 1`] = `
"Created 'build/Markdown/Open Source.html'
Created 'build/The Naval Treaty.html'
Created 'build/The Red Headed League.html'
Created 'build/a/Silver Blaze.html'
Created 'build/a/b/The Adventure of the Six Napoleans.html'
Created 'build/c/The Adventure of the Speckled Band.html'"
`;

exports[`end to end test use input option on a file 1`] = `"Created 'build/The Naval Treaty.html'"`;

exports[`end to end test use invalid css file 1`] = `"'src/index.ts' is not a css file"`;

exports[`end to end test use invalid language  1`] = `"Unsupported HTML language"`;

exports[`end to end test use recursive option 1`] = `
"Created 'build/Open Source.html'
Created 'build/Silver Blaze.html'
Created 'build/The Adventure of the Six Napoleans.html'
Created 'build/The Adventure of the Speckled Band.html'
Created 'build/The Naval Treaty.html'
Created 'build/The Red Headed League.html'"
`;

exports[`end to end test use recursive option on a file 1`] = `"Created 'build/The Naval Treaty.html'"`;

exports[`end to end test use relative option 1`] = `
"Created 'build/The Naval Treaty.html'
Created 'build/The Red Headed League.html'"
`;
